### PR TITLE
Make module class names all uppercased and session names all capitalised

### LINKED ISSUE #275: [PE-D][Tester D] Difficulty in testing `help` Command
While it's good that `help` opens up a help window if it's unable to open the browser, it's a bit difficult to test since we need to be connected to the internet during the actual PE. Furthermore, I tried to disconnect my internet but it seems like it's still redirecting me to the webpage, so I'm not sure if this is intended, so this might actually be a `type.FunctionalityBug` since I'm not exactly sure what an 'unsuccessful' attempt would mean.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: XenonShawn/ped#1

### LINKED ISSUE #349: Bug: `view` displays sessions based on case used during `grade`
If we have a session `s/Lab1` and we grade a student with `grade 1 s/laB1 g/100`, the result from `view` will display the session name as `s/laB1` instead of `s/Lab1`.

### PR DESCRIPTION
**Pull request by njxue**
- Does not change collection of Sessions into UniqueList. This is done separately in #342 
- Module Class names are now in all caps (i.e. `addc c/cs1101s` adds a module class `CS1101S`)
- Session names are capitalised (first letter of each word is uppercased) (i.e. `adds s/lab1` adds a session `Lab1`. `adds s/_foo_  bAR` adds a session `_foo_ Bar`). This should resolve #349 if it still isn't

**Reduced code duplication** (methods in `CommandUtil`):
- `requireFocusMode` checks if model is in focus mode
- `requireSessionExists` and `requireModuleClassExists` checks if session/module class exists

**UG:**
- Updated UG showing images with correct (uppercased) class names and (capitalised) session names
- Other grammatical/punctuation issues
- Resolved #275

### CODE REVIEW

File: src/main/java/seedu/taassist/logic/commands/CommandUtil.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import seedu.taassist.commons.core.Messages;
+import seedu.taassist.logic.commands.exceptions.CommandException;
+import seedu.taassist.model.Model;
+import seedu.taassist.model.moduleclass.ModuleClass;
+import seedu.taassist.model.session.Session;
+
+/**
+ * Contains utility methods used by commands.
+ */
+public class CommandUtil {
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/logic/commands/CommandUtil.java:16]: Great CodeQuality update!

File: src/main/java/seedu/taassist/logic/commands/CommandUtil.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     * Checks if {@code session} exists in {@code moduleClass}.
+     *
+     * @param session Session object to check for existence.
+     * @param moduleClass ModuleClass object to find the given {@code session}
+     * @throws CommandException if {@code session} does not exist in {@code moduleClass}.
+     */
+    public static void requireSessionExists(Session session, ModuleClass moduleClass) throws CommandException {
+        requireAllNonNull(moduleClass, session);
+        if (!moduleClass.hasSession(session)) {
+            throw new CommandException(String.format(MESSAGE_SESSION_DOES_NOT_EXIST, session.getSessionName(),
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/logic/commands/CommandUtil.java]: Qualify the string constants with `Messages` for consistency. Similarly bellow.

File: src/test/java/seedu/taassist/model/stubs/ModelStubWithNoModuleClass.java [hunk 3]
```diff
@@ -53,6 +47,11 @@ public ObservableList<Student> getStudentList() {
         return new UniqueList<Student>().asUnmodifiableObservableList();
     }
 
+    @Override
+    public ObservableList<Student> getFilteredStudentList() {
+        return new UniqueList<Student>().asUnmodifiableObservableList();
```
Comment (RezwanArefin01) [inline src/test/java/seedu/taassist/model/stubs/ModelStubWithNoModuleClass.java]:
```suggestion
        return FXCollections.<Student>emptyObservableList();
```
Check if java can infer the `<Student>` argument (it should be I think).

File: src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    @Test
+    public void constructor_nullSession_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DeletesCommand(null));
+    }
+
+    @Test
+    public void execute_notInFocusMode_throwsCommandException() {
+        ModelStubNeverInFocusMode modelStub = new ModelStubNeverInFocusMode();
+        Session validSession = new SessionBuilder().withName(VALID_SESSION_LAB1).build();
+        Set<Session> validSessions = new HashSet<>(List.of(validSession));
```
Comment (RezwanArefin01) [inline src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java]: Can we just use `LAB_1` here? 

I think the string constants were for parsing related tests. In command testing we should be using the `Typical*` constants directly. 

Some other command tests may also be updated.

File: src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    @Test
+    public void execute_existingSessionInFocusMode_success() throws CommandException {
+        ModelStubWithModuleClassWithSessions modelStub = new ModelStubWithModuleClassWithSessions();
+        assert !modelStub.getFocusedClass().getSessions().isEmpty();
+        Session existingSession = modelStub.getFocusedClass().getSessions().get(0);
+        Set<Session> existingSessions = new HashSet<>(List.of(existingSession));
+        DeletesCommand command = new DeletesCommand(existingSessions);
+        CommandResult commandResult = command.execute(modelStub);
+        assertEquals(DeletesCommand.getCommandMessage(existingSessions), commandResult.getFeedbackToUser());
```
Comment (RezwanArefin01) [inline src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java]: Don't we need to verify that `removeSessions` method was called with the correct sessions?

Similar issue in other tests bellow.

File: src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        model.setModuleClass(focusedClass, focusedClass.addSession(LAB_1).addSession(TUTORIAL_1));
+        assert model.getFocusedClass().hasSession(LAB_1);
+        assert model.getFocusedClass().hasSession(TUTORIAL_1);
+        Set<Session> sessions = new HashSet<>(List.of(LAB_1, TUTORIAL_1));
+        DeletesCommand command = new DeletesCommand(sessions);
+        assertCommandSuccess(command, model, DeletesCommand.getCommandMessage(sessions), expectedModel);
+    }
+
+    @Test
+    public void execute_nonExistingSession_success() {
```
Comment (RezwanArefin01) [inline src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java]: This is a failure test.

File: src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        assertEquals(DeletesCommand.getCommandMessage(existingSessions), commandResult.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_nonExistingSessionInFocusMode_success() {
+        ModelStubWithModuleClassWithSessions modelStub = new ModelStubWithModuleClassWithSessions();
+        assert !modelStub.getFocusedClass().hasSession(TUTORIAL_1);
+        Set<Session> nonExistingSessions = new HashSet<>(List.of(TUTORIAL_1));
+        DeletesCommand command = new DeletesCommand(nonExistingSessions);
+        assertThrows(CommandException.class, () -> command.execute(modelStub));
```
Comment (RezwanArefin01) [inline src/test/java/seedu/taassist/logic/commands/DeletesCommandTest.java:99]: Need to check the message.

**Review by RezwanArefin01 — CHANGES_REQUESTED** [review]
Suggester some consistency related changes.

File: src/test/java/seedu/taassist/model/stubs/ModelStubWithNoModuleClass.java (hunk 3 above)
Comment (njxue) [reply src/test/java/seedu/taassist/model/stubs/ModelStubWithNoModuleClass.java:52]: Nope, I tried. Java was not able to infer

**Review by RezwanArefin01 — APPROVED** [review]
Looks good to merge!

**Pull request merged**